packages/core-app-api/src/apis/implementations/auth/bitbucket/BitbucketAuth.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import BitbucketIcon from '@material-ui/icons/FormatBold';
+import { DefaultAuthConnector } from '../../../../lib/AuthConnector';
+import { BitbucketSession } from './types';
+import {
+  OAuthApi,
+  SessionApi,
+  SessionState,
+  ProfileInfo,
+  BackstageIdentity,
+  AuthRequestOptions,
+  Observable,
+} from '@backstage/core-plugin-api';
+import { SessionManager } from '../../../../lib/AuthSessionManager/types';
+import {
+  AuthSessionStore,
+  StaticAuthSessionManager,
+} from '../../../../lib/AuthSessionManager';
+import { OAuthApiCreateOptions } from '../types';
+
+export type BitbucketAuthResponse = {
+  providerInfo: {
+    accessToken: string;
+    scope: string;
+    expiresInSeconds: number;
+  };
+  profile: ProfileInfo;
+  backstageIdentity: BackstageIdentity;
+};
+
+const DEFAULT_PROVIDER = {
+  id: 'bitbucket',
+  title: 'Bitbucket',
+  icon: BitbucketIcon,
+};
+
+class BitbucketAuth implements OAuthApi, SessionApi {
```

Comment:
It should be possible to implement this like the other standard OAuth providers like e.g. the google one, i.e. we just need the static `create` method that in turn calls out to `OAuth2.create`

Reply:
I've followed the google example and refactored, thanks!